# **INDIAN STARTUP ECOSYSTEM**

## Project Description
We embark on a journey of discovery as we leverage our data analysis expertise to uncover the untapped potential within the Indian startup ecosystem. This project is designed to not only decode the numbers but to distill insights that will guide our team towards a successful foray into this dynamic market.

## Scope of Work

- Conduct a thorough exploration of datasets, dissecting funding patterns, sectoral nuances, and geographical hotspots in the Indian startup landscap
- Analyze funding received by startups in india from 2018 to 2021



## Hypothesis 

**Null Hypothesis (H0)**: There is no significant relationship between funding and the sector  

**Alternative Hypothesis (H1)**: There is a significant relationship between funding and the sector

## Questions 
1. How does funding vary across different industry sectors in India?
2. How does funding vary with the loaction of the start-ups
3. What is the relationship between the amount of funding and the stage of the company?
4. How have funding trends evolved between 2018 and 2021?
5. What are the most attractive sectors for investors?
6. Does the location of the company influence its sector?




# **DATA EXPLORATION, DATA UNDERSTANDING and DATA ANALYSIS**

In [1]:
# Load libraries
# Database connnection
import pyodbc     
from dotenv import dotenv_values    #import the dotenv_values function from the dotenv package

# Data manipulation
import pandas as pd
import numpy as np

# Data visualization
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

from scipy.stats import f_oneway

# filter warnings
import warnings 
warnings.filterwarnings('ignore')

# **1. Loading and Inspection of Data**

## **1.1 Loading data from the SQL server**

In [2]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("SERVER")
database = environment_variables.get("DATABASE")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")

In [3]:
# Create a connection string

connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"
    
connection = pyodbc.connect(connection_string)



In [4]:
# sql query to get 2020 data. 
query_2020="SELECT * FROM dbo.LP1_startup_funding2020"

# sql query to get 2021 data. 
query_2021="SELECT * FROM dbo.LP1_startup_funding2021"

In [5]:
    # load 2021 data
data_2021=pd.read_sql(query_2021,connection)

    # load 2020 data
data_2020=pd.read_sql(query_2020,connection)

## **1.2 Loading CSV Files**

In [6]:
# load 2019 data
data_2019=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2019.csv')

    # load 2018 data
data_2018=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2018.csv')


# **2.Exploratory Data Analaysis(EDA)**

## **2.0.  2018 Dataset EDA**

In [7]:
# preview the rows and columns for the 2018 dataset
data_2018.sample(5)

Company Name          Industry Round/Series  \
404  BIGPHI TECHNOLOGIES PVT. LTD.           Fashion        Angel   
64                       Sensibull  Trading Platform         Seed   
174                       Deftouch                 —         Seed   
140                    OTO Capital        Automotive         Seed   
66                        MedCords          Wellness         Seed   

          Amount                     Location  \
404     $900,000  Bangalore, Karnataka, India   
64   ₹25,000,000  Bangalore, Karnataka, India   
174            —  Bangalore, Karnataka, India   
140  ₹50,000,000   Mumbai, Maharashtra, India   
66   ₹26,400,000     Pune, Maharashtra, India   

                                         About Company  
404  BIGPHI technologies is the India's first techn...  
64                      Investment Management Platform  
174  Deftouch is a mobile game development company ...  
140  OTO Capital is a platform that allows you to o...  
66   Medcords is India’s first holistic healthcare ...

In [8]:
# checking for number of columns and rows
print (data_2018.shape)
print(f"There are {data_2018.shape[0]} rows, and {data_2018.shape[1]} columns")

(526, 6)
There are 526 rows, and 6 columns


In [9]:
# checking info
data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [10]:
# Describing the data
data_2018.describe()

Company Name Industry Round/Series Amount  \
count               526      526          526    526   
unique              525      405           21    198   
top     TheCollegeFever        —         Seed      —   
freq                  2       30          280    148   

                           Location  \
count                           526   
unique                           50   
top     Bangalore, Karnataka, India   
freq                            102   

                                            About Company  
count                                                 526  
unique                                                524  
top     TheCollegeFever is a hub for fun, fiesta and f...  
freq                                                    2

**Findings**  
- TheCollegeFever company was the most common company  

- Seed series was the most preffered  

- most companies were based in Bangalore location  

In [11]:
# checking for duplicates
print("There are ",data_2018.duplicated().sum(),"duplicate(s)")

There are  1 duplicate(s)


In [12]:
# Checking for missing values
data_2018.isnull().sum()

Company Name     0
Industry         0
Round/Series     0
Amount           0
Location         0
About Company    0
dtype: int64

## **2.1. 2019 EDA**

In [13]:
# get a sample of 2019 dataset
data_2019.sample(5)

Company/Brand  Founded HeadQuarter       Sector  \
2             Eduisfun      NaN      Mumbai       Edtech   
64             Moms Co      NaN   New Delhi   E-commerce   
23  DROR Labs Pvt. Ltd   2018.0       Delhi  Safety tech   
0       Bombay Shaving      NaN         NaN    Ecommerce   
87              Spinny   2015.0       Delhi   Automobile   

                                         What it does  \
2             It aims to make learning fun via games.   
64  It is into mother and baby care-focused consum...   
23  It uses technology to create a trust-based net...   
0          Provides a range of male grooming products   
87                                Online car retailer   

                                             Founders  \
2                                       Jatin Solanki   
64                                      Malika Sadani   
23                     Dhiraj Naubhar, Dheeraj Bansal   
0                                  Shantanu Deshpande   
87  Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...   

                                             Investor    Amount($)  \
2      Deepak Parekh, Amitabh Bachchan, Piyush Pandey  $28,000,000   
64               Saama Capital, DSG Consumer Partners   $5,000,000   
23                          Inflection Point Ventures  Undisclosed   
0                                Sixth Sense Ventures   $6,300,000   
87  Norwest Venture Partners, General Catalyst, Fu...  $50,000,000   

            Stage  
2   Fresh funding  
64       Series B  
23            NaN  
0             NaN  
87            NaN

In [14]:
# checking for number of columns and rows
print (data_2019.shape)
print(f"There are {data_2019.shape[0]} rows, and {data_2019.shape[1]} columns")

(89, 9)
There are 89 rows, and 9 columns


In [15]:
# checking for duplicates
print("There are ",data_2019.duplicated().sum(),"duplicate(s)")

There are  0 duplicate(s)


In [16]:
# Checking for nulls
data_2019.isnull().sum()

Company/Brand     0
Founded          29
HeadQuarter      19
Sector            5
What it does      0
Founders          3
Investor          0
Amount($)         0
Stage            46
dtype: int64

In [17]:
# checking for datatypes in the different columns
data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [18]:
# performing descriptive analysis
data_2019.describe(include='all')

Company/Brand      Founded HeadQuarter  Sector      What it does  \
count             89    60.000000          70      84                89   
unique            87          NaN          17      52                88   
top         Kratikal          NaN   Bangalore  Edtech  Online meat shop   
freq               2          NaN          21       7                 2   
mean             NaN  2014.533333         NaN     NaN               NaN   
std              NaN     2.937003         NaN     NaN               NaN   
min              NaN  2004.000000         NaN     NaN               NaN   
25%              NaN  2013.000000         NaN     NaN               NaN   
50%              NaN  2015.000000         NaN     NaN               NaN   
75%              NaN  2016.250000         NaN     NaN               NaN   
max              NaN  2019.000000         NaN     NaN               NaN   

                          Founders     Investor    Amount($)     Stage  
count                           86           89           89        43  
unique                          85           86           50        15  
top     Vivek Gupta, Abhay Hanjura  Undisclosed  Undisclosed  Series A  
freq                             2            3           12        10  
mean                           NaN          NaN          NaN       NaN  
std                            NaN          NaN          NaN       NaN  
min                            NaN          NaN          NaN       NaN  
25%                            NaN          NaN          NaN       NaN  
50%                            NaN          NaN          NaN       NaN  
75%                            NaN          NaN          NaN       NaN  
max                            NaN          NaN          NaN       NaN

**Findings**
- Kratikal company was the most common company

- Most companies had their headquarters based in Bangalore

- Majority of the companies are involved in the Edtech sector

## **2.2. 2020 EDA**

In [19]:
# get a sample of 2020 dataset
data_2020.sample(5)

Company_Brand  Founded HeadQuarter        Sector  \
294  Treebo Hotels   2015.0   Bangalore   Hospitality   
708        GroCurv   2019.0    Gurugram           B2B   
588   JetSynthesys   2012.0        Pune  Digital tech   
805        Fingpay   2016.0      Mumbai       Fintech   
559   Ather Energy   2013.0   Bangalore    Automobile   

                                          What_it_does  \
294  Treebo is India's tech-enabled, asset-light af...   
708  It offers a wide range of business related ser...   
588       Digital entertainment and technology company   
805                         Provides payment solutions   
559                      Electric scooter manufacturer   

                                             Founders  \
294  Kadam Jeet Jain, Rahul Chaudhary, Sidharth Gupta   
708                 Venkatesh Rangachari, Vipul Gupta   
588                                       RajanNavani   
805                   Pratyush Halen, Anuraag Agrawal   
559                         Tarun Mehta, Swapnil Jain   

                                         Investor      Amount       Stage  \
294  Ramesh Mangaleswaran, Saikiran Krishnamurthy   6000000.0        None   
708                   Unicorn Investments Pte Ltd         NaN  Seed Round   
588          Adar Poonawalla, Kris Gopalakrishnan  40000000.0        None   
805                               IvyCap Ventures   3500000.0    Series A   
559                                 Hero MotoCorp  11241000.0    Series C   

    column10  
294     None  
708     None  
588     None  
805     None  
559     None

In [20]:
#checking for number of columns and rows
print (data_2020.shape)
print(f"There are {data_2020.shape[0]} rows, and {data_2020.shape[1]} columns")

(1055, 10)
There are 1055 rows, and 10 columns


In [21]:
# checking for duplicates
print("There are ",data_2020.duplicated().sum(),"duplicate(s)")

There are  3 duplicate(s)


In [22]:
# Checking for nulls
data_2020.isnull().sum()

Company_Brand       0
Founded           213
HeadQuarter        94
Sector             13
What_it_does        0
Founders           12
Investor           38
Amount            254
Stage             464
column10         1053
dtype: int64

In [23]:
# checking for datatypes in the different columns
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.6+ KB


In [24]:
# Describe the data
data_2020.describe(include='all')

Company_Brand      Founded HeadQuarter   Sector  \
count           1055   842.000000         961     1042   
unique           905          NaN          77      302   
top            Nykaa          NaN   Bangalore  Fintech   
freq               6          NaN         317       80   
mean             NaN  2015.363420         NaN      NaN   
std              NaN     4.097909         NaN      NaN   
min              NaN  1973.000000         NaN      NaN   
25%              NaN  2014.000000         NaN      NaN   
50%              NaN  2016.000000         NaN      NaN   
75%              NaN  2018.000000         NaN      NaN   
max              NaN  2020.000000         NaN      NaN   

                            What_it_does       Founders           Investor  \
count                               1055           1043               1017   
unique                               990            927                848   
top     Provides online learning classes  Falguni Nayar  Venture Catalysts   
freq                                   4              6                 20   
mean                                 NaN            NaN                NaN   
std                                  NaN            NaN                NaN   
min                                  NaN            NaN                NaN   
25%                                  NaN            NaN                NaN   
50%                                  NaN            NaN                NaN   
75%                                  NaN            NaN                NaN   
max                                  NaN            NaN                NaN   

              Amount     Stage  column10  
count   8.010000e+02       591         2  
unique           NaN        42         2  
top              NaN  Series A  Pre-Seed  
freq             NaN        96         1  
mean    1.130430e+08       NaN       NaN  
std     2.476635e+09       NaN       NaN  
min     1.270000e+04       NaN       NaN  
25%     1.000000e+06       NaN       NaN  
50%     3.000000e+06       NaN       NaN  
75%     1.100000e+07       NaN       NaN  
max     7.000000e+10       NaN       NaN

**Findings**  
- Nykaa company was the most common company  

- Majority of the companies had their headquarter at Bangalore  

- The average Amount invested was 113042969.54

## **2.3. 2021 EDA**

In [25]:
# get a sample of 2021 dataset
data_2021.sample(5)

Company_Brand  Founded HeadQuarter                             Sector  \
944         Svish   2020.0    Gurugram                     Consumer Goods   
151        Rapido   2015.0   Bangalore  Information Technology & Services   
99        NeoDocs   2020.0      Mumbai                         Healthtech   
515        Vegrow   2020.0   Bangalore                            Farming   
524   Sea6 Energy   2010.0   Bangalore                      Biotechnology   

                                          What_it_does  \
944                       D2C personal hygiene startup   
151  Rapido is a bike taxi service spread widely ac...   
99   Neodocs is a part of India's prominent Social ...   
515  Vegrow is in the business of fresh fruits & ve...   
524  Sea6 Energy envisions a future where the abund...   

                                              Founders  \
944                             Ishan, Jaideep Mahajan   
151      Aravind Sanka, Pavan Guntupalli, SR Rishikesh   
99          Nikunj Malpani, Anurag Meena, Pratik Lodha   
515  Praneeth Kumar, Shobhit Jain, Mrudhukar Batchu...   
524  Sailaja Nori, Nelson Vadassery, Sowmya Balendi...   

                                           Investor        Amount     Stage  
944                                           Nueva       $400000      Seed  
151              Shell Ventures, Yamaha, Kunal Shah     $52000000      None  
99           Y Combinator, 9Unicorns, Titan Capital  $Undisclosed      None  
515  Lightspeed Venture Partners, Elevation Capital     $13000000      None  
524                                      Aqua-Spark      $9000000  Series B

In [26]:
# get the number of rows and columns for the datasets
print(f'{data_2021.shape}')

print(f"The 2021 dataset has {data_2021.shape[0]} rows and {data_2021.shape[1]} Columns")

(1209, 9)
The 2021 dataset has 1209 rows and 9 Columns


In [27]:
# checking for duplicates
print("There are ",data_2021.duplicated().sum(),"duplicate(s)")

There are  19 duplicate(s)


In [28]:
#  Checking for nulls
data_2021.isnull().sum()

Company_Brand      0
Founded            1
HeadQuarter        1
Sector             0
What_it_does       0
Founders           4
Investor          62
Amount             3
Stage            428
dtype: int64

In [29]:
# checking for datatypes in the different columns
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


In [30]:
# Describe the dataset
data_2021.describe(include='all')

Company_Brand      Founded HeadQuarter   Sector  \
count           1209  1208.000000        1208     1209   
unique          1033          NaN          70      254   
top         BharatPe          NaN   Bangalore  FinTech   
freq               8          NaN         426      122   
mean             NaN  2016.655629         NaN      NaN   
std              NaN     4.517364         NaN      NaN   
min              NaN  1963.000000         NaN      NaN   
25%              NaN  2015.000000         NaN      NaN   
50%              NaN  2018.000000         NaN      NaN   
75%              NaN  2020.000000         NaN      NaN   
max              NaN  2021.000000         NaN      NaN   

                                             What_it_does  \
count                                                1209   
unique                                               1143   
top     BharatPe develops a QR code-based payment app ...   
freq                                                    4   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

                                Founders                   Investor  \
count                               1205                       1147   
unique                              1095                        937   
top     Ashneer Grover, Shashvat Nakrani  Inflection Point Ventures   
freq                                   7                         24   
mean                                 NaN                        NaN   
std                                  NaN                        NaN   
min                                  NaN                        NaN   
25%                                  NaN                        NaN   
50%                                  NaN                        NaN   
75%                                  NaN                        NaN   
max                                  NaN                        NaN   

              Amount Stage  
count           1206   781  
unique           278    31  
top     $Undisclosed  Seed  
freq              73   246  
mean             NaN   NaN  
std              NaN   NaN  
min              NaN   NaN  
25%              NaN   NaN  
50%              NaN   NaN  
75%              NaN   NaN  
max              NaN   NaN

**Findings**  
- BharatPe is the most common company  

- Most companies invested in the FinTech sector  

- Majority of the companies were found in Bangalore  


**Overall Observations:**  
**Issues with the data**

1. There is a discrepancy in the naming conventions between the columns in the 2018 and 2019 datasets compared to the 2020 and 2021 datasets.

2. The 2018 dataset exhibits some missing columns, contributing to an incomplete representation of the data.

3. Conversely, the 2020 dataset contains an additional column that appears to be extraneous and does not serve a meaningful purpose in our analysis.

**Course of Action:**

1. **Missing Column Engineering for 2018:**
   - We will address the absence of certain columns in the 2018 dataset by employing data engineering techniques to create and populate the missing columns, ensuring a comprehensive and consistent dataset.

2. **Column Name Standardization:**
   - To establish uniformity and coherence across all datasets, we will embark on a column renaming process for the 2018 and 2019 datasets. This action aims to align the naming conventions with those observed in the 2020 and 2021 datasets, facilitating seamless data integration and analysis.

3. **Extraneous Column Removal in 2020:**
   - The redundant column identified in the 2020 dataset will be removed, streamlining the dataset and eliminating unnecessary elements that do not contribute to the overall analysis objectives.

These actions collectively enhance the integrity, consistency, and completeness of the dataset, paving the way for a more robust and coherent analytical process.
ical process.







# **3. Data Cleaning and Preparation**

## **3.1. Convert Indian Rupee values in the 2018 data to dollars**


In [31]:
# Replace '₹', commas, '—', and "''" in 'Amount' column
data_2018['Amount'] = data_2018['Amount'].str.replace(',', '').str.replace('—', '').str.replace("''",'').replace('', np.nan)

# Conditionally apply multiplication only where '₹' is present
mask = data_2018['Amount'].str.contains('₹', na=False)
data_2018.loc[mask, 'Amount'] = data_2018.loc[mask, 'Amount'].str.replace('₹', '').astype(float) * 0.0146

### **Add year data was collected**

- There is need to add a column that represents the year each dataset was collected. This will help with handling the datasets after merging the dataframes 


In [32]:
# add year when data was collected column to every dataset
data_2018['year'] = pd.to_datetime('2018', format='%Y').year
data_2019['year'] = pd.to_datetime('2019', format='%Y').year
data_2020['year'] = pd.to_datetime('2020', format='%Y').year
data_2021['year'] = pd.to_datetime('2021', format='%Y').year

In [33]:
def concat_dataframes(data,data1, data2, data3):
    """
    Concatenates four dataframes along the row axis.

    Parameters:
    data (pandas.DataFrame): The first dataframe to concatenate.
    data1 (pandas.DataFrame): The second dataframe to concatenate.
    data2 (pandas.DataFrame): The third dataframe to concatenate.
    data3 (pandas.DataFrame): The fourth dataframe to concatenate.

    Returns:
    pandas.DataFrame: The concatenated dataframe.
    """
    # Rename columns to make them consistent across dataframes
    df1 = data.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded','HeadQuarter': 'headquarter','Sector': 'sector','What_it_does': 'about_company','Founders':'founders','Investor':'investor','Amount':'amount','Stage':'series', 'year' : 'year'})
    df2 = data1.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded','HeadQuarter': 'headquarter','Sector': 'sector','What_it_does': 'about_company','Founders':'founders','Investor':'investor','Amount':'amount','Stage':'series','year' : 'year'})
    df3 = data2.rename(columns={'Company/Brand': 'company_brand', 'Founded': 'founded','HeadQuarter': 'headquarter','Sector': 'sector','What it does': 'about_company','Founders':'founders','Investor':'investor','Amount($)':'amount','Stage':'series','year' : 'year'})
    df4 = data3.rename(columns={'Company Name': 'company_brand','Location': 'headquarter','Industry': 'sector','About Company': 'about_company','Amount':'amount','Round/Series':'series','year' : 'year'})

    # Concatenate dataframes along the row axis
    result = pd.concat([df1, df2, df3,df4], axis=0).reset_index(drop=True)

    return result

## **Concatenate the dataframes**

In [34]:
# concating the dataframes
df= concat_dataframes(data_2021,data_2020,data_2019,data_2018)

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2879 entries, 0 to 2878
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_brand  2879 non-null   object 
 1   founded        2110 non-null   float64
 2   headquarter    2765 non-null   object 
 3   sector         2861 non-null   object 
 4   about_company  2879 non-null   object 
 5   founders       2334 non-null   object 
 6   investor       2253 non-null   object 
 7   amount         2474 non-null   object 
 8   series         1941 non-null   object 
 9   year           2879 non-null   int64  
 10  column10       2 non-null      object 
dtypes: float64(1), int64(1), object(9)
memory usage: 247.5+ KB


In [36]:
#Drop the extreneous column 10
df.drop('column10', axis=1, inplace= True)

## **Cleaning 'Amount' column**

**Notes**  
- Remove all currency signs  

- Remove all other umwanted characters, words and symbols  

- Fill the nulls uning interpolate method 

- Convert the column from object to float

In [37]:
df['amount']=df['amount'].replace('\$','',regex=True)
df['amount']=df['amount'].replace(['Undisclosed','undisclosed',None],np.NAN)
df['headquarter']=df['headquarter'].str.strip('\t#REF!')
df['amount']=df['amount'].replace('\,','',regex=True)
df['amount']=df['amount'].replace(",","")



In [38]:
# fix index 242
df.at[242, 'headquarter'] = None
df.at[242, 'sector'] = 'Pharmaceuticals'
df.at[242, 'founders'] = 'Varun Khanna'
df.at[242, 'investor'] = 'Morgan Stanley Private Equity Asia'
df.at[242, 'amount'] = '22000000'
df.at[242, 'series'] = 'Series C'
df.at[242, 'about_company'] = 'Development and Manufacturing'

# drop the duplicate row
df = df.drop(256, axis=0, errors='ignore')

# fix index 1148
df.at[1148, 'sector'] = 'Logistics'
df.at[1148, 'founders'] ='Basant Kumar' 
df.at[1148, 'investor'] =None
df.at[1148, 'amount'] = 1000000
df.at[1148, 'series'] = 'Seed'
 
# fix index 257
df.at[257, 'sector'] = 'Transport'
df.at[257, 'founders'] ='Anshuman Maheshwary'
df.at[257, 'investor'] =None
df.at[257, 'amount'] = 5000000
df.at[257, 'series'] = 'Seed'

# fix index 98
df.at[98, 'sector'] = 'Computer Games'
df.at[98, 'founders'] ='Pritesh Kumar'
df.at[98, 'investor'] =None
df.at[98, 'amount'] = 1200000
df.at[98, 'series'] = 'Seed'

# drop the duplicate column
df = df.drop(111, axis=0, errors='ignore')

# fix index 677
df.at[677, 'amount'] = 1000000
df.at[677, 'series'] = 'Angel'

# fix index 545
df.at[545, 'investor'] =None
df.at[545, 'amount'] = 1000000
df.at[545, 'series'] = 'Pre-series A'

# fix index 538
df.at[538, 'investor'] =None
df.at[538, 'amount'] = 300000
df.at[538, 'series'] = 'Pre-Seed'

# fix index 551
df.at[551,'amount']=300000
df.at[551,'series']=None

# fix index 674
df.at[674,'amount']=6000000
df.at[674,'series']=None

# fix index 2531
df.at[2531,'amount']=None
df.at[2531,'series']=None


In [39]:
# change the amount column to numeric
df['amount']=pd.to_numeric(df['amount'])

## **Cleaning data_year column**

**Notes**  


- Convert data type to period


In [40]:
# Convert the data_year column to date
df['year']=pd.to_datetime(df['year'], format='%Y')
# df['year']=df['year'].dt.to_period('y')
df['founded']=pd.to_datetime(df['founded']).dt.year

In [41]:
# check for nulls and duplicated
print(f"There are {df['year'].isna().sum()} Null values in the 'year' column")

There are 0 Null values in the 'year' column


In [42]:
df['year'].info()

<class 'pandas.core.series.Series'>
Index: 2877 entries, 0 to 2878
Series name: year
Non-Null Count  Dtype         
--------------  -----         
2877 non-null   datetime64[ns]
dtypes: datetime64[ns](1)
memory usage: 109.5 KB


## **Cleaning 'founded' column**

**Notes**
- Handle nulls by populating with the 'bfill' method



In [43]:
print(f"There are {df['founded'].isna().sum()} Null values in the 'founded' column")

There are 769 Null values in the 'founded' column


**Notes**  
- There are 769 null values in the 'founded' column.  

- Since dropping the nulls will lead to a significant loss of our data, Backward fill will be used to fill the null values

**COURSE OF ACTION**

- We will be cleaning the missing values by fill them using interpolate method which is suitable for time series data  

- We will also be converting the data type from float to datetime for purpose of our analysis.

In [44]:
# Fill the nulls
df["founded"].interpolate(method='linear',inplace= True)

print(f"There are {df['founded'].isna().sum()} missing values")

There are 0 missing values


In [45]:
# Convert to datetime
df['founded'] = pd.to_datetime(df['founded'], format='%Y')

# Convert to period
df['founded'] = df['founded'].dt.to_period('Y')


In [46]:
df['founded'].info()

<class 'pandas.core.series.Series'>
Index: 2877 entries, 0 to 2878
Series name: founded
Non-Null Count  Dtype        
--------------  -----        
2877 non-null   period[Y-DEC]
dtypes: period[Y-DEC](1)
memory usage: 109.5 KB


In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2877 entries, 0 to 2878
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   company_brand  2877 non-null   object        
 1   founded        2877 non-null   period[Y-DEC] 
 2   headquarter    2762 non-null   object        
 3   sector         2859 non-null   object        
 4   about_company  2877 non-null   object        
 5   founders       2332 non-null   object        
 6   investor       2246 non-null   object        
 7   amount         2320 non-null   float64       
 8   series         1941 non-null   object        
 9   year           2877 non-null   datetime64[ns]
dtypes: datetime64[ns](1), float64(1), object(7), period[Y-DEC](1)
memory usage: 311.8+ KB


## **Cleaning the 'founders' column**

In [48]:
# Remove unwanted characters
df['founders'] = df['founders'].replace(['...', np.nan], np.NaN)

# Check the number of NaN values in the 'founders' column
nan_count = df['founders'].isna().sum()

print(nan_count)

545


In [49]:
df['founders'].info()

<class 'pandas.core.series.Series'>
Index: 2877 entries, 0 to 2878
Series name: founders
Non-Null Count  Dtype 
--------------  ----- 
2332 non-null   object
dtypes: object(1)
memory usage: 109.5+ KB


## **Cleaning **Stage** column**

Startups start with pre-seed, progress through seed, Series A, Series B, etc., securing resources for development and strategies. Additional rounds like Series C or D may follow. External funding at each stage fuels growth toward the venture's full potential.

**Pre-Seed Funding**  
Entrepreneurial idea in early development; small funds needed; limited informal channels for raising funds.

**Seed Funding**  
First official equity funding; investors provide funds for equity ownership.

**Series A Financing**  
First venture capital round; developed product, consistent revenue, long-term profit plan.

**Series B Financing**  
For established startups; substantial user base and revenue; funding for expansion.

**Series C and Beyond**  
Optional rounds for final push before IPO or unmet objectives; Series C is the third venture capital round.

**Initial Public Offering (IPO)**  
Process of offering corporate shares to the public; used for funding or divestment.

link: https://www.startupindia.gov.in/content/sih/en/funding.html

In [50]:
# fix index 551
df.at[551,'amount']=300000
df.at[551,'series']=None

# fix index 674
df.at[674,'amount']=6000000
df.at[674,'series']=None

# fix index 2531
df.at[2531,'amount']=None
df.at[2531,'series']=None

In [51]:
# Standardize funding stages in the 'stage' column
df['series'] = df['series'].replace(['Series A', 'Seies A', 'Series A-1', 'Series A2', 'Series A+', 'Series A+'], 'Series A')
df['series'] = df['series'].replace(['Pre-seed', 'Pre-seed Round', 'Pre seed Round','Non-equity Assistance', 'Pre seed round'], 'Pre-Seed Stage')
df['series'] = df['series'].replace(['Pre series A', 'Pre-series A', 'Pre Series A', 'Pre series A1', 'Pre-series A1', 'Pre- series A'], 'Pre series A')
df['series'] = df['series'].replace(['Series B', 'Series B+', 'Series B2', 'Mid series','Series B3'], 'Series B')
df['series'] = df['series'].replace(['Series C', 'Series C', 'Series C, D','Series C', 'Private Equity','PE', 'Post-IPO Equity','Series D', 'Series E', 'Series F', 'Series G', 'Series H', 'Series I','Series D1','Series F2', 'Series F1'], 'Series C and Beyond')
df['series'] = df['series'].replace(['Venture - Series Unknown', None], 'unknown')
df['series'] = df['series'].replace(['Corporate Round','Undisclosed','Secondary Market','Pre-series','Post series A','Pre-series B','Pre-Series B','Pre series B','Pre-series C','Pre series C''Grant','Debt','Debt Financing','Post-IPO Debt','Bridge','Bridge Round','Fresh funding','Funding Round','Edge'], 'Other Stages')
df['series'] = df['series'].replace(['Seed','Seed funding','Pre-Seed','Angel', 'Angel Round','Seed fund', 'Seed round', 'Seed A','Seed Funding', 'Seed Round & Series A', 'Series E2', 'Seed Round','Seed Investment','Seed+','Early seed'],'Seed Stage')

In [52]:
df['series'].unique()

array(['Pre series A', 'unknown', 'Series C and Beyond', 'Seed Stage',
       'Series B', 'Pre-Seed Stage', 'Series A', 'Other Stages',
       'Pre series C', 'Grant'], dtype=object)

In [53]:
# strip off characters 
df['series'] = df['series'].str.strip('\t#REF!')

## **Cleaning the Sector Column**

In [54]:
# Get the first sentence of every list
df['sector']=df['sector'].str.split(",").str[0]

In [55]:

# Replace the sector values using the provided formula
df['sector'] = df['sector'].replace(['AI startup', 'AI Chatbot', 'AI Company', 'AI platform', 'AI Platform', 'AI Startup', 'Artificial Intelligence'], 'Artificial Intelligence')
df['sector'] = df['sector'].replace(['EdTech', 'EdTech Startup', 'EdtTech', 'Edtech'], 'Education Technology')
df['sector'] = df['sector'].replace(['B2B E-commerce', 'E-commerce', 'B2B Ecommerce','Commerce','B2B marketplace', 'B2B Marketplace'], 'Business-to-Business E-commerce')
df['sector'] = df['sector'].replace(['FinTech'], 'Fintech')
df['sector'] = df['sector'].replace(['Home services'], 'Home Services')
df['sector'] = df['sector'].replace(['HealthTech', 'Healthtech', 'Healtcare', 'HeathTech','Alternative Medicine', 'Health care','Helathcare', 'Healthcare', 'Health', 'Hospital & Health Care', 'Heathcare'], 'Healthcare')
df['sector'] = df['sector'].replace(['Tech Startup', 'Tech startup', 'IT startup', 'Technology Startup'], 'Technology Startup')
df['sector'] = df['sector'].replace(['B2B service'], 'Business-to-Business Services')
df['sector'] = df['sector'].replace(['Renewable Energy'], 'Renewable Energy')
df['sector'] = df['sector'].replace(['Electronics'], 'Electronics')
df['sector'] = df['sector'].replace(['Food & Beverages', 'Food and Beverages', 'Food', 'Food Production', 'Food Industry', 'Food tech', 'Foodtech', 'FoodTech', 'Food Startup', 'Food & tech', 'Food diet', 'Food Processing', 'Food and Beverage', 'Food delivery', 'Food Delivery', 'Food & Logistics', 'Food devlivery', 'QSR startup'], 'Food and Beverages')
df['sector'] = df['sector'].replace(['Aeorspace', 'Aerospace', 'Aviation', 'Aviation & Aerospace', 'Aero company'], 'Aerospace')
df['sector'] = df['sector'].replace(['Deep Tech', 'Deeptech'], 'Deep Technology')
df['sector'] = df['sector'].replace(['Dating', 'Dating app'], 'Dating Services')
df['sector'] = df['sector'].replace(['Gaming', 'Computer Games', 'Mobile Games', 'Games', 'E-Sports', 'Gaming startup'], 'Gaming')
df['sector'] = df['sector'].replace(['Robotics', 'Robotics & AI'], 'Robotics')
df['sector'] = df['sector'].replace(['Retail', 'Reatil startup', 'Retail Startup', 'Retail Tech'], 'Retail')
df['sector'] = df['sector'].replace(['Oil and Energy', 'Oil & Energy'], 'Oil and Energy')
df['sector'] = df['sector'].replace(['AgriTech', 'Agritech', 'Agritech startup', 'AgTech', 'Agri tech', 'B2B Agritech', 'Agricultural Technology'], 'Agricultural Technology')
df['sector'] = df['sector'].replace(['Telecommuncation', 'Telecommunication', 'Telecommunications'], 'Telecommunications')
df['sector'] = df['sector'].replace(['Milk startup', 'Dairy startup', 'Dairy'], 'Dairy')
df['sector'] = df['sector'].replace(['IT', 'Information Technology', 'Information Technology & Services', 'Information Technology Company', 'Information Technology and Services', 'IT company'], 'Information Technology')
df['sector'] = df['sector'].replace(['Logistics', 'Logistics & Supply Chain', 'Logistics and Supply Chain'], 'Logistics and Supply Chain')
df['sector'] = df['sector'].replace(['Hospitality'], 'Hospitality')
df['sector'] = df['sector'].replace(['Fashion', 'Fashion & Lifestyle', 'Fashion and lifestyle', 'Fashion startup', 'Fashion Tech', 'Apparel & Fashion', 'Apparel and Fashion', 'Clothing', 'D2C Fashion', 'D2C jewellery'], 'Fashion and Lifestyle')
df['sector'] = df['sector'].replace(['Marketing', 'Marketing & Advertising', 'Marketing and Advertising', 'MarTech', 'Marketing Technology', 'Advertisement', 'Advertising', 'Marketing company', 'Marketing startup'], 'Marketing and Advertising')
df['sector'] = df['sector'].replace(['Transportation', 'Transport', 'Mobility', 'Mobility/Transport', 'Mobility tech', 'Auto-tech', 'Logitech', 'Transport & Rentals', 'Transport Automation', 'Transportation-as-a-Service Startup', 'TaaS startup', 'Automotive and Rentals', 'IoT/Automobile', 'Automobile Technology', 'Mobility', 'Last Mile Transportation', 'Micro-mobiity'], 'Transportation and Mobility')
df['sector'] = df['sector'].replace(['LegalTech', 'Legaltech', 'Legal Services', 'Legal tech', 'Legaltech'], 'Legal Technology')
df['sector'] = df['sector'].replace(['Automotive', 'Automobile', 'Automotive Startup', 'Automotive company', 'Car Trade', 'Automotive tech', 'Vehicle repair startup', 'Car Service', 'Luxury car startup'], 'Automotive')
df['sector'] = df['sector'].replace(['SaaS startup', 'SaaS', 'SaaS platform', 'Software-as-a-Service', 'Software-as-a-Service Platform', 'Software-as-a-Service Startup', 'SaaS\xa0\xa0startup'], 'Software-as-a-Service')
df['sector'] = df['sector'].replace(['Fantasy sports', 'Fantasy Sports'], 'Fantasy Sports')
df['sector'] = df['sector'].replace(['Video communication', 'Video platform', 'Video', 'Video sharing platform', 'Video streaming platform', 'Video personalization'], 'Video')
df['sector'] = df['sector'].replace(['Social Media', 'Social network', 'Social Networking', 'Social commerce', 'Social community', 'Social Network', 'Social e-commerce', 'Social media', 'Social platform', 'Social audio'], 'Social Media')
df['sector'] = df['sector'].replace(['Skill development', 'Skills Development', 'Professional Training & Coaching', 'Professional Training and Coaching'], 'Skills Development and Professional Training')
df['sector'] = df['sector'].replace(['Rental', 'Rental Services', 'Rental space', 'Bike Rental', 'Furniture Rental'], 'Rental Services')
df['sector'] = df['sector'].replace(['Recruitment', 'Job discovery platform', 'Job portal', 'Staffing & Recruiting', 'Staffing and Recruiting', 'Recruitment startup'], 'Recruitment and Staffing')
df['sector'] = df['sector'].replace(['Sports', 'SportsTech', 'Sports Technology', 'sports','Sports startup'], 'Sports and Sports Technology')
df['sector'] = df['sector'].replace(['Consumer Goods', 'Consumer goods', 'Consumer software', 'Consumer service', 'Consumer Services', 'Consumer', 'Consumer Applications', 'Consumer appliances', 'Consumer Electronics'], 'Consumer Goods and Services')
df['sector'] = df['sector'].replace(['HRTech', 'HR Tech', 'HR Tech startup', 'Human Resources', 'Human Resources Technology', 'HRTech', 'HR tech', 'HR'], 'Human Resources and HR Technology')
df['sector'] = df['sector'].replace(['Wine & Spirits'], 'Wine and Spirits')
df['sector'] = df['sector'].replace(['Mechanical & Industrial Engineering', 'Mechanical and Industrial Engineering', 'Mechanical Or Industrial Engineering'], 'Mechanical and Industrial Engineering')
df['sector'] = df['sector'].replace(['Spiritual', 'Linguistic Spiritual'], 'Spirituality')
df['sector'] = df['sector'].replace(['Financial Services', 'Banking', 'Capital Markets','Finance', 'Finance company', 'Investment Banking', 'Investment Management', 'Finance', 'Finance company', 'Financial Technology', 'FinTech', 'Digital mortgage', 'Online financial service', 'Online credit management startup', 'Trading platform', 'Virtual Banking'], 'Financial Services')
df['sector'] = df['sector'].replace(['Industrial Automation'], 'Industrial Automation')
df['sector'] = df['sector'].replace(['IoT', 'Internet of Things', 'IoT platform', 'IoT startup', 'Internet of Things Platform', 'Internet of Things Startup'], 'Internet of Things')
df['sector'] = df['sector'].replace(['Computer software', 'Computer Software', 'Software', 'Software company', 'Software Startup'], 'Computer Software')
df['sector'] = df['sector'].replace(['Furniture', 'Home Decor', 'Interior Design'], 'Furniture and Home Decor')
df['sector'] = df['sector'].replace(['Location Analytics'], 'Location Analytics')
df['sector'] = df['sector'].replace(['Media', 'Online Media', 'Podcast', 'Podcasting', 'Content publishing', 'Publication', 'Digital Media', 'Media and Entertainment', 'Broadcasting', 'News', 'Digital Entertainment', 'Media Tech', 'Visual Media', 'Online Portals', 'Content creation', 'Content Management', 'Content Marktplace'], 'Media and Entertainment')
df['sector'] = df['sector'].replace(['Consumer Electronics'], 'Consumer Electronics')
df['sector'] = df['sector'].replace(['Tobacco'], 'Tobacco')
df['sector'] = df['sector'].replace(['Insuretech', 'Insurance Technology', 'Insurtech', 'InsureTech', 'Insurance'], 'Insurance')
df['sector'] = df['sector'].replace(['MLOps platform'], 'Machine Learning Operations')
df['sector'] = df['sector'].replace(['Venture Capital', 'Venture Capital & Private Equity', 'Venture Capital and Private Equity', 'Venture capitalist'], 'Venture Capital')
df['sector'] = df['sector'].replace(['Pet care', 'Veterinary', 'Veterinary Services'], 'Pet Care and Veterinary Services')
df['sector'] = df['sector'].replace(['Drone', 'Drone Technology'], 'Drone Technology')
df['sector'] = df['sector'].replace(['Wholesale'], 'Wholesale')
df['sector'] = df['sector'].replace(['Construction', 'Construction tech'], 'Construction and Construction Technology')
df['sector'] = df['sector'].replace(['E-learning', 'E-learning', 'Education Management'], 'E-learning and Education Management')
df['sector'] = df['sector'].replace(['OTT', 'Over-the-Top Media Services'], 'Over-the-Top Media Services')
df['sector'] = df['sector'].replace(['Computer & Network Security', 'Computer and Network Security', 'Cybersecurity'], 'Computer and Network Security')
df['sector'] = df['sector'].replace(['Social network', 'Social Networking', 'Social Network', 'Social community', 'Social platform'], 'Social Networking and Community Platforms')
df['sector'] = df['sector'].replace(['Music', 'Music Streaming'], 'Music and Music Streaming')
df['sector'] = df['sector'].replace(['Hauz Khas'], 'Hauz Khas')
df['sector'] = df['sector'].replace(['Pharmaceuticals', 'Pharmaceutical', 'Pharma', 'Biopharma', 'Pharmaceuticals'], 'Pharmaceuticals')
df['sector'] = df['sector'].replace(['Cosmetics', 'Beauty products', 'Personal Care', 'Beauty & wellness', 'Personal care startup', 'Skincare startup'], 'Cosmetics, Personal Care, and Beauty Products')
df['sector'] = df['sector'].replace(['Real Estate', 'Real estate', 'Commercial Real Estate', 'Housing Marketplace', 'Proptech', 'Property Technology'], 'Real Estate and Property Technology')
df['sector'] = df['sector'].replace(['Digital platform', 'Digital Platform'], 'Digital Platforms')
df['sector'] = df['sector'].replace(['NFT Marketplace', 'Non-Fungible Token Marketplace', 'NFT', 'Non-Fungible Tokens'], 'NFT and Non-Fungible Tokens')
df['sector'] = df['sector'].replace(['SpaceTech', 'Spacetech', 'Space Technology', 'Defense & Space'], 'Space Technology and Defense')
df['sector'] = df['sector'].replace(['Fishery'], 'Fishery')
df['sector'] = df['sector'].replace(['Renewables & Environment', 'Renewable Energy and Environmental Services', 'Environmental Services', 'Environmental service', 'Clean Technology', 'CleanTech', 'Pollution control equiptment'], 'Renewable Energy, Environmental Services, and Clean Technology')
df['sector'] = df['sector'].replace(['Online storytelling'], 'Online Storytelling')
df['sector'] = df['sector'].replace(['CRM', 'Customer Relationship Management'], 'Customer Relationship Management')
df['sector'] = df['sector'].replace(['Nutrition', 'Nutrition sector', 'Nutrition tech', 'Nutrition Tech', 'Food & Nutrition'], 'Nutrition and Nutrition Technology')
df['sector'] = df['sector'].replace(['D2C', 'Direct-to-Consumer', 'D2C Business', 'D2C startup', 'D2C Fashion', 'D2C jewellery'], 'Direct-to-Consumer')
df['sector'] = df['sector'].replace(['Innovation Management', 'Innovation management'], 'Innovation Management')
df['sector'] = df['sector'].replace(['Community platform', 'Community', 'Communities'], 'Community Platforms and Communities')
df['sector'] = df['sector'].replace(['Networking', 'Networking platform'], 'Networking and Networking Platforms')
df['sector'] = df['sector'].replace(['AR startup', 'Augmented Reality Startup', 'AR platform', 'Augmented reality', 'Augmented Reality', 'AR/VR', 'AR/VR startup'], 'Augmented Reality and AR/VR')
df['sector'] = df['sector'].replace(['Content publishing', 'Content commerce', 'Content creation', 'Content Management', 'Content Marktplace'], 'Content')
df['sector'] = df['sector'].replace(['Rental space'], 'Rental Space')
df['sector'] = df['sector'].replace(['Tourism'], 'Tourism')
df['sector'] = df['sector'].replace(['FemTech', 'Femtech'], 'Femtech')
df['sector'] = df['sector'].replace(['Cultural'], 'Cultural')
df['sector'] = df['sector'].replace(['Co-working', 'Co-working Startup', 'Coworking'], 'Co-working')
df['sector'] = df['sector'].replace(['Supply chain platform', 'Supply Chain Platform', 'Logistics & Supply Chain', 'Logistics and Supply Chain', 'B2B Supply Chain'], 'Supply Chain and Logistics')
df['sector'] = df['sector'].replace(['Crypto', 'Cryptocurrency'], 'Cryptocurrency')
df['sector'] = df['sector'].replace(['Clothing'], 'Clothing')
df['sector'] = df['sector'].replace(['Analytics', 'Location Analytics', 'Data Analytics', 'Data Science', 'Data Intelligence'], 'Analytics and Data Science')
df['sector'] = df['sector'].replace(['IoT platform', 'Internet of Things Platform'], 'Internet of Things Platform')
df['sector'] = df['sector'].replace(['Defense & Space', 'Defense and Space'], 'Defense and Space')

## **Cleaning HeadQuater Column**

In [56]:
# Get the first location from every list
df['headquarter']=df['headquarter'].str.split(",").str[0]

In [57]:
df['headquarter']=df['headquarter'].replace({'Bengaluru': 'Bangalore', 'Banglore': 'Bangalore', 'Gurugram': 'Gurgaon', 'Hyderebad': 'Hyderabad', 
                                      'New Delhi': 'Delhi', 'Ahmadabad': 'Ahmedabad', 'Ernakulam': 'Cochin', 'Telugana': 'Telangana',
                                      'Rajastan': 'Rajasthan', 'San Franciscao': 'San Francisco', 'Samsitpur': 'Samastipur', 'Santra': 'Samtra',
                                      'Rajsamand': 'Rajasthan', 'Kerala': 'Kochi','The Nilgiris': 'Nilgiris', 'Gurugram\t#REF!': 'Gurgaon', 
                                      'California': 'San Francisco', 'San Francisco Bay Area': 'San Francisco', 'Hyderebad': 'Hyderabad',
                                      'Online Media\t#REF!': 'Unknown','Pharmaceuticals\t#REF!': 'Unknown',
                                      'Information Technology & Services':'Unknown' ,'Small Towns': 'Unknown','Orissia': 'Odisha', 
                                      'Santra': 'Samtra', 'Vadodara': 'Vadodara', 'Food & Beverages': 'Unknown', 'Bangaldesh': 'Bangladesh',}) 


In [58]:
df.head()

company_brand founded headquarter                           sector  \
0  Unbox Robotics    1970   Bangalore          Artificial Intelligence   
1          upGrad    1970      Mumbai             Education Technology   
2     Lead School    1970      Mumbai             Education Technology   
3         Bizongo    1970      Mumbai  Business-to-Business E-commerce   
4        FypMoney    1970     Gurgaon                          Fintech   

                                       about_company  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        investor       amount  \
0                    BEENEXT, Entrepreneur First    1200000.0   
1       Unilazer Ventures, IIFL Asset Management  120000000.0   
2               GSV Ventures, Westbridge Capital   30000000.0   
3                         CDC Group, IDG Capital   51000000.0   
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    2000000.0   

                series       year  
0         Pre series A 2021-01-01  
1              unknown 2021-01-01  
2  Series C and Beyond 2021-01-01  
3  Series C and Beyond 2021-01-01  
4           Seed Stage 2021-01-01

In [59]:
# check for duplicates
df[df.duplicated()]


company_brand founded      headquarter  \
107            Curefoods    1970        Bangalore   
109             Bewakoof    1970           Mumbai   
117       Advantage Club    1970           Mumbai   
119               Ruptok    1970            Delhi   
243             Trinkerr    1970        Bangalore   
244                Zorro    1970          Gurgaon   
245        Ultraviolette    1970        Bangalore   
246           NephroPlus    1970        Hyderabad   
247              Unremot    1970        Bangalore   
248          FanAnywhere    1970        Bangalore   
249           PingoLearn    1970             Pune   
250                 Spry    1970           Mumbai   
251              Enmovil    1970        Hyderabad   
252        ASQI Advisors    1970           Mumbai   
253   Insurance Samadhan    1970            Delhi   
254      Evenflow Brands    1970           Mumbai   
255           MasterChow    1970  ood & Beverages   
1354           Krimanshi    1970          Jodhpur   
1414               Nykaa    1970           Mumbai   
1571              Byju’s    1970        Bangalore   
2701     TheCollegeFever    1970        Bangalore   

                                             sector  \
107                              Food and Beverages   
109                           Fashion and Lifestyle   
117               Human Resources and HR Technology   
119                                         Fintech   
243                              Financial Services   
244                                    Social Media   
245                                      Automotive   
246                                      Healthcare   
247                          Information Technology   
248                              Financial Services   
249             E-learning and Education Management   
250                       Music and Music Streaming   
251                          Information Technology   
252                              Financial Services   
253                                       Insurance   
254                     Consumer Goods and Services   
255                                       Hauz Khas   
1354                          Biotechnology company   
1414  Cosmetics, Personal Care, and Beauty Products   
1571                           Education Technology   
2701                                Brand Marketing   

                                          about_company  \
107   Healthy & nutritious foods and cold pressed ju...   
109   Bewakoof is a lifestyle fashion brand that mak...   
117   Advantage Club is India's largest employee eng...   
119   Ruptok fintech Pvt. Ltd. is an online gold loa...   
243   Trinkerr is India's first social trading platf...   
244                Pseudonymous social network platform   
245   Create and Inspire the future of sustainable u...   
246   A vision and passion of redefining healthcare ...   
247       Unremot is a personal office for consultants!   
248                              Celebrity NFT platform   
249   PingoLearn offers language learning courses wi...   
250   THE BEST ROYALTY-FREE MUSIC TO CREATE TRENDING...   
251   Enmovil delivers a Business Intelligence Platf...   
252   Bringing Blockchain technology intro mainstrea...   
253   The insurance industry will help in representa...   
254   Identify and partner with the upcoming eCommer...   
255                 A ready-to-cook Asian cuisine brand   
1354  Krimanshi aims to increase rural income by imp...   
1414  Nykaa is an online marketplace for different b...   
1571  An Indian educational technology and online tu...   
2701  TheCollegeFever is a hub for fun, fiesta and f...   

                                           founders  \
107                                    Ankit Nagori   
109                                Prabhkiran Singh   
117              Sourabh Deorah, Smiti Bhatt Deorah   
119                                     Ankur Gupta   
243                 Manvendra Singh, Gaurav Agarwal   
244 

In [60]:
# drop duplicates
df.drop_duplicates(keep='first', inplace=True)


In [61]:
# check for duplicates
print(f" There are {df.duplicated().sum()} duplicates")

 There are 0 duplicates


In [62]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2856 entries, 0 to 2878
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   company_brand  2856 non-null   object        
 1   founded        2856 non-null   period[Y-DEC] 
 2   headquarter    2741 non-null   object        
 3   sector         2838 non-null   object        
 4   about_company  2856 non-null   object        
 5   founders       2312 non-null   object        
 6   investor       2227 non-null   object        
 7   amount         2301 non-null   float64       
 8   series         2856 non-null   object        
 9   year           2856 non-null   datetime64[ns]
dtypes: datetime64[ns](1), float64(1), object(7), period[Y-DEC](1)
memory usage: 245.4+ KB


# **4. Hypothesis**

**Null Hypothesis (H0)**: There is no significant relationship between funding and the sector  

**Alternative Hypothesis (H1)**: There is a significant relationship between funding and the sector

In [63]:
# Perform one-way ANOVA
result = f_oneway(*[df['amount'][df['sector'] == sector] for sector in df['sector'].unique()])

# Output the results
print(f"F-statistic: {result.statistic}")
print(f"P-value: {result.pvalue}")

# Set the significance level
alpha = 0.05

# Compare p-value with the significance level
if result.pvalue < alpha:
    print("Reject the null hypothesis. There is a significant relationship between funding amount and the sector.")
else:
    print("Fail to reject the null hypothesis. There is no significant relationship between funding amount and the sector.")

F-statistic: nan
P-value: nan
Fail to reject the null hypothesis. There is no significant relationship between funding amount and the sector.


# **5. Analytical Questions**

## Questions 
1. How does funding vary across different industry sectors in India?
2. How does funding vary with the loaction of the start-ups
3. What is the relationship between the amount of funding and the stage of the company?
4. How have funding trends evolved between 2018 and 2021?
5. What are the most attractive sectors for investors?
6. Does the location of the company influence its sector?

In [64]:
df.head(3)

company_brand founded headquarter                   sector  \
0  Unbox Robotics    1970   Bangalore  Artificial Intelligence   
1          upGrad    1970      Mumbai     Education Technology   
2     Lead School    1970      Mumbai     Education Technology   

                                       about_company  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   

                                            founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   

                                   investor       amount               series  \
0               BEENEXT, Entrepreneur First    1200000.0         Pre series A   
1  Unilazer Ventures, IIFL Asset Management  120000000.0              unknown   
2          GSV Ventures, Westbridge Capital   30000000.0  Series C and Beyond   

        year  
0 2021-01-01  
1 2021-01-01  
2 2021-01-01

## **4. How have funding trends evolved between 2018 and 2021?**


### **Funding Trend by Year**

In [65]:
# Plot a line graph to display the trend over the year
funding_trend=df.groupby('year')['amount'].sum()
funding_trend=pd.DataFrame(funding_trend)
# Convert the DataFrame to long format for Plotly Express
# Convert the DataFrame to long format for Plotly Express
funding_trend_long = funding_trend.reset_index()

# Create a line plot using Plotly Express
fig = px.line(funding_trend_long, x='year', y='amount', markers=True, title='Indian Startup Funding Trend from 2018-2021',
              labels={'amount': 'Total Amount Funded in Billion Dollars'})

# Show the plot
fig.show()


#### **Observations**  
- There is slight decrease and then a sudden increase  in the Indian Startup funding from 2019 to 2021
- The steep upward trajectory suggests an increasing trend in funding

#### **Funding Trend by Top Locations**

In [67]:
from matplotlib.dates import YearLocator
df['year'] = pd.to_datetime(df['year'], format='%Y')
location_trend = df.groupby(['headquarter', 'year'])['amount'].sum().sort_values(ascending=False)
top_locations = location_trend.groupby('headquarter').sum().nlargest(4).index
# Filter the original DataFrame for the top 5 locations
location_trend_top4 = location_trend[location_trend.index.get_level_values('headquarter').isin(top_locations)].reset_index()
# Set 'year' as the index for the grouped data
# Reset the index to use 'year' as a column
location_trend_top4_reset = location_trend_top4.reset_index()

# Create a line plot using Plotly Express
fig = px.line(location_trend_top4_reset, x='year', y='amount', color='headquarter',
              markers=True, title='Top 4 Locations Funding Trend Over the Years',
              labels={'amount': 'Total Funded Amount'}, line_dash='headquarter')

# Show the plot
fig.show()
plt.show()

#### **Observations**  
- All four locations have see a substantial increase in funding from 2019 to 2021, however, Mumbai funding grew sharply compared to the other three i.e Bangalore Delhi and Gurgaon